src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -6337,7 +6543,88 @@ private IOperation VisitNoneOperationExpression(IOperation operation)
         {
-            return VisitNoneOperation(operation, captureIdForResult);
+            AssertContainingContextIsForThisCreation(operation, _currentInterpolatedStringHandlerContext);
+            switch (operation.PlaceholderKind)
+            {
+                case InterpolatedStringArgumentPlaceholderKind.TrailingValidityArgument:
+                    if (_currentInterpolatedStringHandlerContext.OutPlaceholder == -1)
+                    {
+                        throw ExceptionUtilities.Unreachable;
+                    }
+
+                    return new FlowCaptureReferenceOperation(_currentInterpolatedStringHandlerContext.OutPlaceholder, operation.Syntax, operation.Type, operation.GetConstantValue(), isInitialization: true);
+
+                case InterpolatedStringArgumentPlaceholderKind.CallsiteReceiver:
+                    if (_currentInterpolatedStringHandlerContext.HasReceiver && tryGetArgumentOrReceiver(-1) is IOperation receiverCapture)
+                    {
+                        Debug.Assert(receiverCapture is IFlowCaptureReferenceOperation);
+                        return OperationCloner.CloneOperation(receiverCapture);
+                    }
+                    else
+                    {
+                        return new InvalidOperation(ImmutableArray<IOperation>.Empty, semanticModel: null, operation.Syntax, operation.Type, operation.GetConstantValue(), isImplicit: true);
+                    }
+
+                case InterpolatedStringArgumentPlaceholderKind.CallsiteArgument:
+                    if (tryGetArgumentOrReceiver(operation.ArgumentIndex) is IOperation argumentCapture)
+                    {
+                        Debug.Assert(argumentCapture is IFlowCaptureReferenceOperation or IDiscardOperation);
+                        return OperationCloner.CloneOperation(argumentCapture);
+                    }
+                    else
+                    {
+                        return new InvalidOperation(ImmutableArray<IOperation>.Empty, semanticModel: null, operation.Syntax, operation.Type, operation.GetConstantValue(), isImplicit: true);
+                    }
+
+                default:
+                    throw ExceptionUtilities.UnexpectedValue(operation.PlaceholderKind);
+            }
+
+            IOperation? tryGetArgumentOrReceiver(int argumentIndex)
+            {
+                int startingStackDepth = _currentInterpolatedStringHandlerContext.StartingStackDepth;
+                if (startingStackDepth >= _evalStack.Count)
+                {
+                    return null;
+                }
+
+                if (argumentIndex == -1)
+                {
+                    Debug.Assert(_currentInterpolatedStringHandlerContext.HasReceiver);
+
+                    // The receiver is always directly at the starting stack depth: there is no additional frame pushed on at
```

Comment:
>// The receiver is always directly at the starting stack depth: there is no additional frame pushed on at [](http://example.com/codeflow?start=20&length=105)

I don't think we have to make this assumption. In fact, the implementation would be more robust if we wouldn't. The solution looks simple, when ```_currentInterpolatedStringHandlerContext.HasReceiver```, increment ```argumentIndex``` and proceed to the loop below.